### PR TITLE
Add DNS services to script

### PR DESCRIPTION
**Pull request by trevharmon**
This commit allows the module to set up basic DNS services for use
during the actual installation. It is expected these settings will
later be managed by Puppet through service-specific classes.

This class allows the configuration of IPv4 and/or IPv6 DNS on one
interface selected by `InterfaceAlias` or `InterfaceIndex`. These
can also be overriden on the command line.

By default, the script will also test the new (or existing) values
to ensure validity before continuing. Administrators also have the
option to not have existing values overriden, if that makes more
sense in their environment. Even in this case, however, the validity
of the existing DNS servers will be checked.

### CODE REVIEW

**Comment by trevharmon** [comment]
Contains all changes of PR #14

**Comment by natemccurdy** [comment]
Please rebase off of master to resolve merge conflicts.

**Comment by natemccurdy** [comment]
@trevharmon Thanks for the PR!

However, to keep the footprint of this script as low as possible by default, can you restructure the code such that configuring DNS is **disabled** by default.

That might also mean adding some logic so that if `override_dns` is set to false all of the other DNS parameters that were added won't do anything.

**Comment by trevharmon** [comment]
Configuring DNS is now disabled by default via the `set_dns_servers` parameter.

File: README.md [hunk 1]
```diff
@@ -90,6 +90,20 @@ $webClient.DownloadFile("https://puppet.company.net:8140/packages/current/instal
 & "$env:temp\install-agent.ps1" -certname win-db001.custom.net -server alternate-puppet-master.custom.net
 ```
 
+#### Enable DNS Settings
+
+By Default, this script will not modify the DNS settings ont he machine, to enable this feature, set the `set_dns` flag to true:
```
Comment (natemccurdy) [inline README.md:95]: minor typo here.

`s/ont he/on the/`

**Comment by trevharmon** [comment]
Will resubmit PR once the code is cleaned up.

**Comment by trevharmon** [comment]
See #18

**Pull request closed without merging**